### ISSUE TITLE
0.17 Not detecting Sensei 310

### ISSUE THREAD

**Issue by NO-ob** [libratbag/libratbag#1501]
**Information**

- `ratbagd` version (`ratbagd --version`): 0.17
- Distribution: Arch
- Kernel version (ex. `uname -srmo`): `Linux 6.4.2-arch1-1 x86_64 GNU/Linux`

Device info (if applicable):
```
$ ratbagctl <device> info
OUTPUT HERE
```

**Describe the bug**
Version 0.17 doesn't detect my Steelseries Sensei 310, restarting the service doesn't effect this. downgrading to version 0.16 works

**To Reproduce**
1. Start the service
2. Open piper
3. No devices found
4. run ratbagctl list
5. No devices found

**Logs**
Please start the daemon with verbosity and reproduce the issue.

First make sure it isn't running already.

0.17 output
```
$ ratbagd --verbose=raw
Initializing libratbag
ratbag debug: New device: SteelSeries SteelSeries Sensei 310 eSports Mouse
ratbag debug: Using data directory '/usr/share/libratbag'
ratbag debug: device assigned driver steelseries
ratbag debug: hidraw info: bus 0x03 vendor 0x1038 product 0x1722
ratbag debug: SteelSeries SteelSeries Sensei 310 eSports Mouse is device '/dev/hidraw1'.
ratbag debug: Parsing HID report descriptor
ratbag debug: Parsing HID report descriptor
ratbag debug: hidraw info: bus 0x03 vendor 0x1038 product 0x1722
ratbag debug: SteelSeries SteelSeries Sensei 310 eSports Mouse is device '/dev/hidraw2'.
ratbag debug: Parsing HID report descriptor
ratbag debug: Parsing HID report descriptor
ratbag debug: hidraw info: bus 0x03 vendor 0x1038 product 0x1722
ratbag debug: SteelSeries SteelSeries Sensei 310 eSports Mouse is device '/dev/hidraw3'.
ratbag debug: Parsing HID report descriptor
ratbag debug: Parsing HID report descriptor
ratbag debug: - HID report ID 01
ratbag debug: - HID report ID 02
ratbag debug: New device: SteelSeries SteelSeries Sensei 310 eSports Mouse
ratbag debug: Using data directory '/usr/share/libratbag'
ratbag debug: device assigned driver steelseries
ratbag debug: hidraw info: bus 0x03 vendor 0x1038 product 0x1722
ratbag debug: SteelSeries SteelSeries Sensei 310 eSports Mouse is device '/dev/hidraw1'.
ratbag debug: Parsing HID report descriptor
ratbag debug: Parsing HID report descriptor
ratbag debug: hidraw info: bus 0x03 vendor 0x1038 product 0x1722
ratbag debug: SteelSeries SteelSeries Sensei 310 eSports Mouse is device '/dev/hidraw2'.
ratbag debug: Parsing HID report descriptor
ratbag debug: Parsing HID report descriptor
ratbag debug: hidraw info: bus 0x03 vendor 0x1038 product 0x1722
ratbag debug: SteelSeries SteelSeries Sensei 310 eSports Mouse is device '/dev/hidraw3'.
ratbag debug: Parsing HID report descriptor
ratbag debug: Parsing HID report descriptor
ratbag debug: - HID report ID 01
ratbag debug: - HID report ID 02
ratbag debug: New device: SteelSeries SteelSeries Sensei 310 eSports Mouse
ratbag debug: Using data directory '/usr/share/libratbag'
ratbag debug: device assigned driver steelseries
ratbag debug: hidraw info: bus 0x03 vendor 0x1038 product 0x1722
ratbag debug: SteelSeries SteelSeries Sensei 310 eSports Mouse is device '/dev/hidraw1'.
ratbag debug: Parsing HID report descriptor
ratbag debug: Parsing HID report descriptor
ratbag debug: hidraw info: bus 0x03 vendor 0x1038 product 0x1722
ratbag debug: SteelSeries SteelSeries Sensei 310 eSports Mouse is device '/dev/hidraw2'.
ratbag debug: Parsing HID report descriptor
ratbag debug: Parsing HID report descriptor
ratbag debug: hidraw info: bus 0x03 vendor 0x1038 product 0x1722
ratbag debug: SteelSeries SteelSeries Sensei 310 eSports Mouse is device '/dev/hidraw3'.
ratbag debug: Parsing HID report descriptor
ratbag debug: Parsing HID report descriptor
ratbag debug: - HID report ID 01
ratbag debug: - HID report ID 02
ratbag debug: New device: SEMITEK USB-HID Gaming Keyboard
ratbag debug: Using data directory '/usr/share/libratbag'
ratbag debug: No data file found for 1ea7:0907
ratbag debug: New device: SEMITEK USB-HID Gaming Keyboard
ratbag debug: Using data directory '/usr/share/libratbag'
ratbag debug: No data file found for 1ea7:0907
ratbag debug: New device: SEMITEK USB-HID Gaming Keyboard
ratbag debug: Using data directory '/usr/share/libratbag'
ratbag debug: No data file found for 1ea7:0907
ratbag debug: New device: GeneralPlus USB Audio Device
ratbag debug: Using data directory '/usr/share/libratbag'
ratbag debug: No data file found for 1b3f:2008
DBus server ready
```

0.16 Output
```
Initializing libratbag
ratbag debug: New device: SteelSeries SteelSeries Sensei 310 eSports Mouse
ratbag debug: Using data directory '/usr/share/libratbag'
ratbag debug: device assigned driver steelseries
ratbag debug: hidraw info: bus 0x03 vendor 0x1038 product 0x1722
ratbag debug: SteelSeries SteelSeries Sensei 310 eSports Mouse is device '/dev/hidraw1'.
ratbag debug: Parsing HID report descriptor
ratbag debug: Parsing HID report descriptor
ratbag debug: hidraw info: bus 0x03 vendor 0x1038 product 0x1722
ratbag debug: SteelSeries SteelSeries Sensei 310 eSports Mouse is device '/dev/hidraw2'.
ratbag debug: Parsing HID report descriptor
ratbag debug: Parsing HID report descriptor
ratbag debug: hidraw info: bus 0x03 vendor 0x1038 product 0x1722
ratbag debug: SteelSeries SteelSeries Sensei 310 eSports Mouse is device '/dev/hidraw3'.
ratbag debug: Parsing HID report descriptor
ratbag debug: Parsing HID report descriptor
ratbag debug: - HID report ID 01
ratbag debug: - HID report ID 02
ratbag debug: hidraw info: bus 0x03 vendor 0x1038 product 0x1722
ratbag debug: SteelSeries SteelSeries Sensei 310 eSports Mouse is device '/dev/hidraw1'.
ratbag debug: Parsing HID report descriptor
ratbag debug: - HID report ID 01
ratbag debug: - HID report ID 02
ratbag debug: Parsing HID report descriptor
ratbag debug: - HID report ID 01
ratbag debug: - HID report ID 02
ratbag raw: output report: 90 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
ratbag raw: input report:  24 01 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
ratbag debug: SteelSeries firmware version 1.36
ratbag raw: output report: 92 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
ratbag raw: input report:  01 01 07 00 0f 00 32 00 39 f8 12 fc 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
ratbag debug: driver match found: SteelSeries
hidraw1: "SteelSeries SteelSeries Sensei 310 eSports Mouse", 1 profiles
ratbag debug: New device: SteelSeries SteelSeries Sensei 310 eSports Mouse
ratbag debug: Using data directory '/usr/share/libratbag'
ratbag debug: device assigned driver steelseries
ratbag debug: hidraw info: bus 0x03 vendor 0x1038 product 0x1722
ratbag debug: SteelSeries SteelSeries Sensei 310 eSports Mouse is device '/dev/hidraw1'.
ratbag debug: Parsing HID report descriptor
ratbag debug: Parsing HID report descriptor
ratbag debug: hidraw info: bus 0x03 vendor 0x1038 product 0x1722
ratbag debug: SteelSeries SteelSeries Sensei 310 eSports Mouse is device '/dev/hidraw2'.
ratbag debug: Parsing HID report descriptor
ratbag debug: Parsing HID report descriptor
ratbag debug: New device: SteelSeries SteelSeries Sensei 310 eSports Mouse
ratbag debug: Using data directory '/usr/share/libratbag'
ratbag debug: device assigned driver steelseries
ratbag debug: hidraw info: bus 0x03 vendor 0x1038 product 0x1722
ratbag debug: SteelSeries SteelSeries Sensei 310 eSports Mouse is device '/dev/hidraw1'.
ratbag debug: Parsing HID report descriptor
ratbag debug: Parsing HID report descriptor
ratbag debug: hidraw info: bus 0x03 vendor 0x1038 product 0x1722
ratbag debug: SteelSeries SteelSeries Sensei 310 eSports Mouse is device '/dev/hidraw2'.
ratbag debug: Parsing HID report descriptor
ratbag debug: Parsing HID report descriptor
ratbag debug: New device: SEMITEK USB-HID Gaming Keyboard
ratbag debug: Using data directory '/usr/share/libratbag'
ratbag debug: No data file found for 1ea7:0907
ratbag debug: New device: SEMITEK USB-HID Gaming Keyboard
ratbag debug: Using data directory '/usr/share/libratbag'
ratbag debug: No data file found for 1ea7:0907
ratbag debug: New device: SEMITEK USB-HID Gaming Keyboard
ratbag debug: Using data directory '/usr/share/libratbag'
ratbag debug: No data file found for 1ea7:0907
ratbag debug: New device: GeneralPlus USB Audio Device
ratbag debug: Using data directory '/usr/share/libratbag'
ratbag debug: No data file found for 1b3f:2008
DBus server ready

```

**Additional context**

**Comment by staticssleever668** [comment]
Hi! This is a duplicate of #1438.
TLDR: there was a faulty device check added in v0.17, which has been reverted in master.
You can use {libratbag,piper}-git from AUR for now.
